datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java

```diff
@@ -38,19 +38,14 @@ public CoordinatorEventBlockingQueue() {
    */
   public synchronized void put(CoordinatorEvent event) {
     LOG.info("Queuing event {} to event queue", event.getType());
-    if (!_eventMap.containsKey(event.getType())) {
+    if (!_eventSet.contains(event)) {
```

Comment:
I almost reached out to you then thought this would prolly be a lot more useful to ask over here for knowledge sharing and wider visibility (think of future maintainers):
1. Why could `put()` be invoked with duplicate events?
2. Why are we rejecting duplicates?

Reply:
1. Heartbeat is added periodically. If the Coordinator Event thread is stuck or slow at processing, we might add another heartbeat event.
2. Reason to reject is if the event is already present in the queue, it will do the exact operation twice (which is unnecessary) and delay the pipeline.